kratos/utilities/parallel_utilities.h

```diff
@@ -152,17 +151,25 @@ class IndexPartition
         }
 
         //here we impose a syncronization and we check the exceptions
+        std::stringstream ss;
         for(int i=0; i<mNchunks; ++i)
         {
             try {
                     runners[i].get();
-            } catch(const std::exception& e) {
-                std::cout << std::endl << "THREAD number: " << i << " caught exception " << e.what() << std::endl;
+            }
+            catch(Exception& e) {
+                ss << std::endl << "THREAD number: " << i << " caught exception " << e.what() << std::endl;
+            } catch(std::exception& e) {
+                ss << std::endl << "THREAD number: " << i << " caught exception " << e.what() << std::endl;
+            } catch(...) {
+                ss << std::endl << "unknown error" << std::endl;
+            }
         }
 
-        }
+        auto msg = ss.str();
+        if(!msg.empty())
```

Comment:
Looks like you are going to pay for this (in performance) even if there were no exceptions. Why not just remove the `try .. catch` block around [line 158](https://github.com/KratosMultiphysics/Kratos/pull/6767/commits/adee928dd67175e752c6387339fc18352b1bb6c9#diff-54fb8cd12185ae6bfc821a13ab7585a2R158) (`runners[i].get()`)?

Reply:
yes, i thought about this. The problem is that if i do that it will only throw the first thread encountering the exception. the way i did you would see the error message of everyone.

do you think that the performance impact is big?

Reply:
You are constructing a `std::ostringstream` in [line 154](https://github.com/KratosMultiphysics/Kratos/pull/6767/commits/adee928dd67175e752c6387339fc18352b1bb6c9#diff-54fb8cd12185ae6bfc821a13ab7585a2R154), and then you are creating a new `std::string` for `msg` in [line 169](https://github.com/KratosMultiphysics/Kratos/pull/6767/commits/adee928dd67175e752c6387339fc18352b1bb6c9#diff-54fb8cd12185ae6bfc821a13ab7585a2R169). I am not sure how expensive is that, but both of these could involve a heap allocation.

> The problem is that if i do that it will only throw the first thread encountering the exception

I think one exception will be enough to kill the program (or to start an investigation). I would not care about lost exceptions form other threads if I get at least one.

Reply:
Another way to think about this: if you include a serial implementation without any threads, you will also only see the first exception. You will not iterate until completion to get other errors, even if there are potentially more of them.

Reply:
ok i will throw on the first error